larq_zoo/literature/meliusnet.py

```diff
@@ -0,0 +1,278 @@
+        x = self.norm(x, name=f"{name}_s1_bn")
+        x = self.act(x, name=f"{name}_s1_relu")
+
+        x = self.group_conv(x, 64, 3, 8, name=f"{name}_s2_groupconv")
+        x = self.norm(x, name=f"{name}_s2_bn")
+        x = self.act(x, name=f"{name}_s2_relu")
+
+        return self.pool(x, name=f"{name}_pool")
+
+    def dense_block(self, x: tf.Tensor, name: str = None) -> tf.Tensor:
+        w = x
+        w = self.norm(w, name=f"{name}_bn")
+        w = self.quant_conv(w, 64, 3, name=f"{name}_binconv")
+        return utils.TFOpLayer(tf.concat, axis=-1, name=f"{name}_concat")([x, w])
+
+    def improvement_block(self, x: tf.Tensor, name: str = None) -> tf.Tensor:
+        w = x
+        w = self.norm(w, name=f"{name}_bn")
+        w = self.quant_conv(w, 64, 3, name=f"{name}_binconv")
+        f_in = int(x.shape[-1])
+        return tf.keras.layers.Lambda(
+            lambda x_: x_[0] + tf.pad(x_[1], [[0, 0], [0, 0], [0, 0], [f_in - 64, 0]]),
+            name=f"{name}_merge",
+        )([x, w])
+
+    def transition_block(
+        self, x: tf.Tensor, filters: int, name: str = None
+    ) -> tf.Tensor:
+        x = self.norm(x, name=f"{name}_bn")
+        x = self.pool(x, name=f"{name}_maxpool")
+        x = self.act(x, name=f"{name}_relu")
+        return tf.keras.layers.Conv2D(
+            filters,
+            1,
+            use_bias=False,
+            kernel_initializer=self.kernel_initializer,
+            name=f"{name}_pw",
+        )(x)
+
+    def block(self, x: tf.Tensor, name: str = None) -> tf.Tensor:
+        x = self.dense_block(x, name=f"{name}_dense")
+        return self.improvement_block(x, name=f"{name}_improve")
+
+    def build(self) -> tf.keras.models.Model:
+        x = self.image_input
+        x = self.group_stem(x, name="stem")
+        for i, (n, f) in enumerate(zip(self.num_blocks, self.transition_features)):
+            for j in range(n):
+                x = self.block(x, f"section_{i}_block_{j}")
+            if f:
+                x = self.transition_block(x, f, f"section_{i}_transition")
+
+        x = self.norm(x, "head_bn")
+        x = self.act(x, "head_relu")
+
+        if self.include_top:
+            x = utils.global_pool(x, name="head_globalpool")
+            x = tf.keras.layers.Dense(
+                self.num_classes,
+                kernel_initializer=self.kernel_initializer,
+                name="head_dense",
+            )(x)
+            x = tf.keras.layers.Activation(
+                "softmax", dtype="float32", name="head_softmax"
+            )(x)
+
+        model = tf.keras.models.Model(
+            inputs=self.image_input, outputs=x, name=self.name
+        )
+
+        if self.weights == "imagenet":
+            if self.include_top:
+                weights_path = self.imagenet_weights_path
+            else:
+                weights_path = self.imagenet_no_top_weights_path
+            model.load_weights(weights_path)
```

Comment:
```suggestion
            model.load_weights(self.imagenet_weights_path if self.include_top else self.imagenet_no_top_weights_path)
```

Reply:
(This'll probably fail the `black` check in CI if you commit it on GitHub, so may want to do it in an editor with Black-on-save.)